R/bambu-extendAnnotations-utilityExtend.R

```diff
@@ -331,31 +353,34 @@ calculateDistToAnnotation <- function(exByTx, exByTxRef, maxDist = 35,
-                                                  exByTxRef, maxDist = 0, type = "any", firstLastSeparate = TRUE,
-                                                  dropRangesByMinLength = FALSE, cutStartEnd = TRUE,
-                                                  ignore.strand = ignore.strand)
-  txToAnTableRest <-
-    genFilteredAnTable(spliceOverlaps_rest, primarySecondaryDist,
-                       exByTx = exByTx, setTMP = setTMP, DistCalculated = FALSE)
-  # (3) find overlaps for remaining reads 
-  setTMPRest <- unique(c(txToAnTableRest$queryHits, setTMP))
-  txToAnTableRestStartEnd <- NULL
-  if (length(exByTx[-setTMPRest])) {
-    spliceOverlaps_restStartEnd <-
-      findSpliceOverlapsByDist(exByTx[-setTMPRest], exByTxRef,
-                               maxDist = 0, type = "any", firstLastSeparate = TRUE,
-                               dropRangesByMinLength = FALSE,
-                               cutStartEnd = FALSE, ignore.strand = ignore.strand)
-    if (length(spliceOverlaps_restStartEnd)) {
-      txToAnTableRestStartEnd <-
-        genFilteredAnTable(spliceOverlaps_restStartEnd,
-                           primarySecondaryDist, exByTx = exByTx,
-                           setTMP = setTMPRest, DistCalculated = TRUE)
+                                                    exByTxRef, maxDist = 0, type = "any", firstLastSeparate = TRUE,
+                                                    dropRangesByMinLength = FALSE, cutStartEnd = TRUE,
+                                                    ignore.strand = ignore.strand)
+  if(length(spliceOverlaps_rest) > 0){
+    txToAnTableRest <-
+      genFilteredAnTable(spliceOverlaps_rest, primarySecondaryDist,
+                        exByTx = exByTx, setTMP = setTMP, DistCalculated = FALSE)
+    # (3) find overlaps for remaining reads 
+    setTMPRest <- unique(c(txToAnTableRest$queryHits, setTMP))
+    txToAnTableRestStartEnd <- NULL
+    if (length(exByTx[-setTMPRest])) {
+      spliceOverlaps_restStartEnd <-
+        findSpliceOverlapsByDist(exByTx[-setTMPRest], exByTxRef,
+                                maxDist = 0, type = "any", firstLastSeparate = TRUE,
+                                dropRangesByMinLength = FALSE,
+                                cutStartEnd = FALSE, ignore.strand = ignore.strand)
+      if (length(spliceOverlaps_restStartEnd)) {
+        txToAnTableRestStartEnd <-
+          genFilteredAnTable(spliceOverlaps_restStartEnd,
+                            primarySecondaryDist, exByTx = exByTx,
+                            setTMP = setTMPRest, DistCalculated = TRUE)
+      }
     }
-  }
-  txToAnTableFiltered <- rbind( txToAnTableFiltered,
-                                txToAnTableRest, txToAnTableRestStartEnd ) %>% ungroup()
+    txToAnTableFiltered <- rbind( txToAnTableFiltered,
+                                  txToAnTableRest, txToAnTableRestStartEnd ) %>% ungroup()
+  } else txToAnTableFiltered %>% ungroup()
```

Comment:
this line might not do anything as there is no assignment here(?)

Reply:
Resolved in 68512b664d9417e1dc218d637a7b9b7be8d04088